lib/sql.js

```diff
@@ -692,7 +692,7 @@ SQLConnector.prototype._replace = function(model, where, data, options, cb) {
   var stmt = this.buildReplace(model, where, data, options);
   this.execute(stmt.sql, stmt.params, options, function(err, info) {
     if (err) return cb(err);
-    if (info.affectedRows === 0) {
+    if (info.affectedRows === 0 || info.rowsAffected === 0) {
```

Comment:
Great idea !! It also works for all the connectors except mssql (which is already fixed). (Y)
 I re-committed.